### PR TITLE
volumetric imaging example

### PR DESCRIPTION
**Pull request by droumis**
volumetric_imaging_header

### CODE REVIEW

File: volumetric_imaging/anaconda-project.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+dependencies: *pkgs
+
+commands:
+  lab:
+    unix: jupyter lab volumetric_imaging.ipynb
+    windows: jupyter lab volumetric_imaging.ipynb
+  notebook:
+    notebook: volumetric_imaging.ipynb
+  dashboard:
```
Comment (maximlt) [inline volumetric_imaging/anaconda-project.yml:41]: There's no `.servable()` in the notebook. Unless you want to deploy it just like that with this newish feature of Panel? That would be a first here!

image

**Review by maximlt — COMMENTED** [review]
The main concern I have with this example at the moment is that it's hard to demonstrate its capabilities:
- Deploying it as a read-only notebook is likely to fail (see comment in the project file)
- The NeuroGlancer views aren't displayed on the website (see screenshots below), which makes it look quite broken

@droumis how about:
- Replacing the actual views in the notebook with screenshots/gifs? (unless the underlying issue can be fixed, I haven't look at all into it)
- Deploying the example as a Panel app?

Chrome:
image

Firefox:
image

File: volumetric_imaging/anaconda-project.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  created: 2024-11-22
+  maintainers:
+  - "droumis"
+  labels:
+  - "panel"
+  categories:
+  - Neuroscience
+  title: "Volumetric Imaging"
+  deployments:
+  - command: notebook
```
[the remark this answers is not in the document]
Comment (droumis) [reply volumetric_imaging/anaconda-project.yml:14]: yep, not much to do except maybe add a 'open in colab badge' or something

File: volumetric_imaging/anaconda-project.yml (hunk 1 above)
Comment (droumis) [reply volumetric_imaging/anaconda-project.yml:41]: It doesn't really make sense as a Panel app in its current form. 

I suggest we add a colab command for notebook use.

Additionally, I'm ok with leaving this newish deploy app thing for panel (does it have a name?).

File: volumetric_imaging/volumetric_imaging.ipynb [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+    "\n",
+    "    def __init__(\n",
+    "        self,\n",
+    "        source=None,\n",
+    "        aspect_ratio=2.75,\n",
+    "        show_state=False,\n",
+    "        load_demo=False,\n",
```
[the remark this answers is not in the document]
Comment (droumis) [reply volumetric_imaging/volumetric_imaging.ipynb]: done

File: volumetric_imaging/volumetric_imaging.ipynb [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+    "        self.local_url_pane = pn.pane.Markdown(\"**Local URL:**\")\n",
+    "\n",
+    "        self.iframe = pn.pane.HTML(\n",
+    "            sizing_mode=\"stretch_both\",\n",
+    "            aspect_ratio=aspect_ratio,\n",
+    "            min_height=800,\n",
+    "            styles={\"resize\": \"both\", \"overflow\": \"hidden\"},\n",
+    "        )\n",
+    "\n",
+    "    def _configure_viewer(self):\n",
```
[the remark this answers is not in the document]
Comment (droumis) [reply volumetric_imaging/volumetric_imaging.ipynb]: done

File: volumetric_imaging/anaconda-project.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+user_fields: [examples_config]
+
+channels:
+- conda-forge
+- nodefaults
+
+packages: &pkgs
+- notebook>=6.5.2,<7 # required
+- python=3.11
+- panel
```
[the remark this answers is not in the document]
Comment (droumis) [reply volumetric_imaging/anaconda-project.yml]: done

**Comment by maximlt** [comment]
@droumis I made a few changes in https://github.com/holoviz-topics/examples/pull/454/commits/6b8e6c77c0d3fac6f11e66b64535d61e70f1170a:

- Realized that we should use `remove-{}` instead of `hide-{}`, `hide` puts the hidden thing in a `details` element (see https://myst-nb.readthedocs.io/en/latest/render/hiding.html), it turns out we tweak the CSS in nbsite to entirely hide `hide-input` cells (https://github.com/holoviz-dev/nbsite/blob/c8f6e244d8b4718f463f53c1530a4fef70d2b0a0/nbsite/_shared_static/notebook.css#L53-L55), I'm not sure why.
- Removed the comment to avoid showing the app (twice), this is to test as much as possible the notebook code on the CI. WIth `remove-output`, the website doesn't show the outputs (with the somewhat broken neuroglancer apps)
- Used the Panel dashboard builder (preview in JupyterLab) to remove a couple of useless cells. FYI I opened https://github.com/holoviz/panel/issues/7546, definitely not a blocker for this PR.

**Comment by jbednar** [comment]
Thanks for this great example!

**Pull request merged**